admin/class-taxonomy-presenter.php

```diff
@@ -0,0 +1,211 @@
+		$help  = $this->get_help( $field, $field_options['description'], $esc_field_name );
+
+		echo $this->parse_row( $label, $help, $field );
+	}
+
+	/**
+	 * Generates the html for the the given field config
+	 *
+	 * @param string $field_type  The fieldtype, e.g: text, checkbox, etc.
+	 * @param string $field_name  The name of the field.
+	 * @param string $field_value The value of the field.
+	 * @param array  $options     Array with additional options.
+	 *
+	 * @return string
+	 */
+	private function get_field($field_type, $field_name, $field_value, array $options) {
+
+		$class = $this->get_class( $options );
+		$field = '';
+
+		switch ( $field_type ) {
+			case 'text' :
+				$field .= '<input name="' . $field_name . '" id="' . $field_name . '" ' . $class . ' type="text" value="' . esc_attr( $field_value ) . '" size="40"/>';
+				break;
+			case 'checkbox' :
+				$field .= '<input name="' . $field_name . '" id="' . $field_name . '" type="checkbox" ' . checked( $field_value ) . '/>';
+				break;
+			case 'textarea' :
+				$rows = 3;
+				if ( ! empty( $options['rows'] ) ) {
+					$rows = $options['rows'];
+				}
+				$field .= '<textarea class="large-text" rows="' . esc_attr( $rows ) . '" id="' . $field_name . '" name="' . $field_name . '">' . esc_textarea( $field_value ) . '</textarea>';
+				break;
+			case 'upload' :
+				$field .= '<input id="' . $field_name . '" type="text" size="36" name="' . $field_name . '" value="' . esc_attr( $field_value ) . '" />';
+				$field .= '<input id="' . $field_name . '_button" class="wpseo_image_upload_button button" type="button" value="Upload Image" />';
+				break;
+			case 'select' :
+				if ( is_array( $options ) && $options !== array() ) {
+					$field .= '<select name="' . $field_name . '" id="' . $field_name . '">';
+
+					$select_options = ( array_key_exists( 'options', $options ) ) ? $options['options'] : $options;
+
+					foreach ( $select_options as $option => $option_label ) {
+						$selected = selected( $option, $field_value, false );
+						$field .= '<option ' . $selected . ' value="' . esc_attr( $option ) . '">' . esc_html( $option_label ) . '</option>';
+					}
+					unset( $option, $option_label, $selected );
+
+					$field .= '</select>';
+				}
+				break;
+			case 'hidden' :
+				$field .= '<input name="' . $field_name . '" id="hidden_' . $field_name . '" type="hidden" value="' . esc_attr( $field_value ) . '" />';
+				break;
+		}
+
+		if ( $field !== '' && ( ! empty( $options['description'] ) && is_string( $options['description'] ) ) ) {
+			$field .= '<p class="description">' . $options['description'] . '</p>';
+		}
+
+		return $field;
+	}
+
+	/**
+	 * Getting the value for given field_name
+	 *
+	 * @param string $field_name The fieldname to get the value for.
+	 *
+	 * @return string
+	 */
+	private function get_field_value( $field_name ) {
+		if ( isset( $this->tax_meta[ $field_name ] ) && $this->tax_meta[ $field_name ] !== '' ) {
+			return $this->tax_meta[ $field_name ];
```

Comment:
why do you need to do the second check? You'll return `''` anyway if it's an empty string. It looks to be that you only need to do an isset check here.